share/wake/lib/core/string.wake

```diff
@@ -341,3 +341,13 @@ export def hashString (x: String): String =
     def hashStr x = prim "hash_str"
 
     hashStr x
+
+export def toLower (x: String): String =
```

Comment:
These need doc comments

Reply:
Added. I didn't feel like adding super detailed docs since I felt like it was mostly self explanatory.